packages/grid/x-data-grid/src/models/gridSlotsComponent.ts

```diff
@@ -82,7 +87,7 @@ export interface GridSlotsComponent extends GridBaseSlots, GridIconSlotsComponen
    * Component rendered for each cell.
    * @default GridCell
    */
-  cell: React.JSXElementConstructor<any>;
+  cell: React.JSXElementConstructor<GridCellProps>;
```

Comment:
Not sure why these are all `any`, but adding types here makes it easier to refactor and not miss issues or create bugs.

Reply:
Yes, I came across this issue as well in https://github.com/mui/mui-x/pull/8067#discussion_r1122007786
At that time, we decided to not go with this change, but I'm all in for doing this, at least for some slots 👍🏻

Minor sidenote: `GridCellProps` does not include [custom props from props overrides](https://mui.com/x/react-data-grid/components/#custom-slot-props-with-typescript). I think this was the reason I tried using `GridSlotsComponentsProps` in the PR linked above.

<img width="691" src="https://github.com/mui/mui-x/assets/13808724/20be87cf-15b9-4e9f-bb51-fd48f90d61d2">

> I commented out the following line in the `GridCellProps` to demonstrate the issue above:
> https://github.com/romgrk/mui-x/blob/da77da1037e6256651a84a32060073b776dddcdb/packages/grid/x-data-grid/src/components/cell/GridCell.tsx#L65

Reply:
It'll be a nice dx improvement if we could replace these any occurrences with proper types.
Could possibly be done separately though?